frontend/src/dispatcher/js-messages.ts

```diff
@@ -124,19 +125,36 @@ export class DisplayFolderTreeStructure extends JsMessage {
 		super();
 	}
 }
-export function newDisplayFolderTreeStructure(input: any): DisplayFolderTreeStructure {
-	const { ptr, len } = input.data_buffer;
-	const wasmMemoryBuffer = (window as any).wasmMemory().buffer;
+
+interface DataBuffer {
+	pointer: number;
+	length: number;
+}
+function isObject(o: unknown): o is Record<string, unknown> {
+	return typeof o === "object" && o !== null;
+}
+function isDataBuffer(o: unknown): o is DataBuffer {
+	return isObject(o) && typeof o.ptr === "number" && typeof o.len === "number";
+}
+function hasDataBuffer(o: unknown): o is { data_buffer: DataBuffer } {
+	return isObject(o) && isDataBuffer(o.data_buffer);
+}
+export function newDisplayFolderTreeStructure(input: unknown, wasm: WasmInstance): DisplayFolderTreeStructure {
+	if (!hasDataBuffer(input)) {
+		throw new TypeError("newDisplayFolderTreeStructure: input does not contain a DataBuffer");
```

Comment:
I do not think it is worth doing runtime checks as we should be able to assume that the rust side will send the correct data

Reply:
Agreed, unless it's likely that a bug can arise during maintenance in the future and this helps explain what's wrong in debugging. If that doesn't seem likely, go ahead and remove it.

Reply:
This is also necessary to satisfy TypeScript. I'm going to leave it for now but feel free to improve it in a later PR.